packages/@lwc/synthetic-shadow/src/faux-shadow/lwc-host.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import { defineProperty } from '@lwc/shared';
+import { setAttribute } from '../env/element';
+
+/*
+ * Using a simple custom attribute is faster based on perf results:
+ * Setting attribute: https://jsperf.com/attribute-vs-data-attribute/1
+ * matches(selector): https://jsperf.com/matches-attribute-vs-data-attribute/1
+ * querySelector(): https://jsperf.com/query-by-attribute-vs-data-attribute/1
+ **/
+export const HostTokenAttributeName = 'lwc-host';
+
+/**
+ * Patching HTMLElement.prototype.$lwcHostToken$ to add a custom attribute to host elements
+ */
+defineProperty(HTMLElement.prototype, '$lwcHostToken$', {
+    get(this: HTMLElement) {
```

Comment:
Considered using a data descriptor, but that would require checking if the property exists using an `in` operator and then invoking the function. Also that function would not return anything.

Reply:
I think what you want here is a `set`, and maybe a get that returns true or false if the attribute is set... for completion.